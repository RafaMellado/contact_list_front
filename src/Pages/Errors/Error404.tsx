import { BsXOctagonFill } from "react-icons/bs";
import { ErrorPage } from "./Components/ErrorPage";

export function Error404() {
  return <ErrorPage code={404} Icon={BsXOctagonFill} />;
}
