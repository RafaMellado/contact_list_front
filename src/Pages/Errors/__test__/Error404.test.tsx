import { Error404 } from "../Error404";
import { act, render } from "@testing-library/react";

describe(`<${Error404.name} />`, () => {
  const factoryComponent = () => render(<Error404 />);

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
