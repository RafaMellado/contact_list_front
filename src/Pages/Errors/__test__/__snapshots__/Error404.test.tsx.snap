// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error404 /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center flex-column"
  >
    <div
      class="d-flex align-items-center flex-column"
    >
      <svg
        class="h1"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.46.146A.5.5 0 0 0 11.107 0H4.893a.5.5 0 0 0-.353.146L.146 4.54A.5.5 0 0 0 0 4.893v6.214a.5.5 0 0 0 .146.353l4.394 4.394a.5.5 0 0 0 .353.146h6.214a.5.5 0 0 0 .353-.146l4.394-4.394a.5.5 0 0 0 .146-.353V4.893a.5.5 0 0 0-.146-.353L11.46.146zm-6.106 4.5L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 1 1 .708-.708z"
        />
      </svg>
      <h2>
        errors.404.title
      </h2>
    </div>
    <h5>
      errors.404.description
    </h5>
    <div
      class="mt-2"
    >
      <button
        class="w-100 ms-2 btn btn-primary"
        data-testid="error-page-btn"
        type="button"
      >
        errors.back
      </button>
    </div>
  </div>
</DocumentFragment>
`;
